ui/src/api.ts

```diff
@@ -123,3 +117,36 @@ async function doGet<T>(get: Promise<AxiosResponse<T>>) {
   }
   return null;
 }
+
+/** 
+ * helper function for slicing arrays.
+ */
+function sliceArray(array: string[], size: number) {
+  var result = [];
+
+  for (var i = 0; i < array.length; i += size) {
+    result.push(array.slice(i, i+size));
+  }
+
+  return result
+}
+
+/** 
+ * Collect array of promises generated based on array of paper ids.
+ */
+function collectPromise(papers:string[], size:number) {
+  var paperChuck = sliceArray(papers, size);
+  var promises = [];
+
+  for (var i = 0; i < paperChuck.length; i++) {
```

Comment:
is this a typo, should this be `paperChunk`?

Reply:
Ooooooops yes!